src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Versioning;
+using NuGet.Packaging.Core;
+
+#if PRODUCT
+using Microsoft.Build.Framework;
+#endif
+
+namespace Microsoft.NET.Build.Tasks
+{
+    /// <summary>
+    /// The following provides the logic to parse the output - artifact.xml.generated by ComposeCache
+    /// </summary>
+    internal class CacheArtifactParser
+    {
+        internal static HashSet<PackageIdentity> Parse(string filterFile)
+        {
+            var listofPackages = new HashSet<PackageIdentity>();
+            var doc = XDocument.Load(filterFile);
+            var rootName = doc.Root.Name;
+#if PRODUCT
+            if (!rootName.LocalName.Equals("CacheArtifacts"))
+            {
+                throw new BuildErrorException(Strings.IncorrectFilterFormat, filterFile);
+            }
+#endif
```

Comment:
I strongly disklike the ifdef precedent, it leads to spaghetti every time. You can leave it for now, though, and I'll look at fixing it by enabling IVTA from tests (#656)